src/_png.cpp

```diff
@@ -328,10 +328,10 @@ static void _read_png_data(PyObject *py_file_obj, png_bytep data, png_size_t len
     Py_ssize_t bufflen;
     if (read_method) {
         result = PyObject_CallFunction(read_method, (char *)"i", length);
-    }
-    if (PyBytes_AsStringAndSize(result, &buffer, &bufflen) == 0) {
-        if (bufflen == (Py_ssize_t)length) {
-            memcpy(data, buffer, length);
+        if (PyBytes_AsStringAndSize(result, &buffer, &bufflen) == 0) {
+            if (bufflen == (Py_ssize_t)length) {
+                memcpy(data, buffer, length);
+            }
```

Comment:
Good point.  We should probably set an exception in that case.  It basically means that it read past the end of the file (which could happen if the file were truncated or something).